### ISSUE TITLE
Global updates breaking DBs of entries. (Mangadex specifically)

### ISSUE THREAD

**Issue by serveral1** [mihonapp/mihon#162]
### Steps to reproduce

1. Add any random entry from mangadex to library
2. Mark all chapters as read
3. Start global update
4. See chapters from entry missing/duplicated/marked as unread

### Expected behavior

Chapter list from the entry to not break after a global update

### Actual behavior

Entry's DB seem to break up after a global update, making some chapters appear as missing/unread/duplicated

### Mihon version

TachiyomiSy 1.10.2 (Based on latest Mihon version)

### Android version

Android 10

### Device

SP-6150

### Other details

A few days ago I made a similar issue on the "other" repo where an issue pertaining a source would be made in, however they closed that issue saying it was an app issue and not something that they could fix (I won't cross link the issue here due to policy reasons but you can find it on my profile activity).

There's some relevant info in that other issue that I don't really want to reiterate. However, since then I've figured out a bit more about what's happening, and it seems to be something related to connection issues when the global update happens (yesterday I did a global update and it went fine, however today I was downloading some stuff on my PC when the global update happened on my phone and issues arose again). Now, my connection isn't exactly the fastest and entries failing to update due to connection issues have happened before, however it was only after the 0.15 update that my manga entries started breaking like this:
![297150348-848e4ae4-21b5-4d67-91b4-e77f48ab95c2](https://github.com/mihonapp/mihon/assets/10162026/648a9d16-2e24-4e9f-b217-8be923929496)
With these kind of errors:
![Screenshot_20240120-142448](https://github.com/mihonapp/mihon/assets/10162026/b83fa4a1-cfd3-4a9c-abf5-0b70cd7c2360)
![Screenshot_20240121-144505](https://github.com/mihonapp/mihon/assets/10162026/26c482a5-7af7-417a-b199-219e5548e35b)

I'm also making the issue here and not on SY's repo because another user (in that other issue I mentioned) also managed to reproduce the behavior on 0.15.3

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an official extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/extensions/issues/new/choose).
- [X] I have gone through the [FAQ](https://mihon.app/docs/faq/general) and [troubleshooting guide](https://mihon.app/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.1](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by BraveSoldat** [comment]
I seem to be having this same bug. At the very least the duplication of chapters has happened to me. I first noticed this on Tachiyomi so I made the switch over to Mihon to see if this happens there too. I still haven't noticed this on Mihon, but the duplicated chapters carried over to Mihon with the backup.

**Comment by Macx6** [comment]
I also have this issue, but even more severe. With global update I notice duplicated/missing chapters, read chapters marked as unread and broken chapter date times resulting in complete mess in my library.

![Screenshot (Feb 18, 2024 11-10-36)](https://github.com/mihonapp/mihon/assets/60160841/c1d3a7ad-8a46-4b92-954b-34c9b61fb6ae)

When I update specific title it fixes date times and missing chapters but not duplicate chapters/unread.

I'm using SY 10.0.2, but update to 10.0.4 didn't fix this. I checked logs but on default log level couldn't find anything specific and I don't think it's related to internet as I think my connection is quite stable and fast enough. I wonder if this is some kind of corruption with library database when global update is performed.

**Comment by serveral1** [comment]
Well, I've pretty much mitigated the issue by disabling the scheduled updates and just doing the global updates myself.

I've noticed however that the download index sometimes just breaks whenever I open up the app after a while, so maybe that is part of the issue as well, though I have no idea why it's just mangadex that is being affected.

**Comment by BraveSoldat** [comment]
This is just an idea but, could this be because we have both Mihon and Tachiyomi installed?
I noticed that even though I set up Tachiyomi to never update (manual update), and Mihon to update every twelve hours, the library kept being updated on Tachiyomi and it never updated on Mihon, unless manually updated.
Perhaps having both apps installed caused some kind of conflict, oerhaps with the extensions, because since I uninstalled Tachi and left only Mihon, it began auto-updating properly.  
Also this issue with repeated chapters stopped altogether.

**Comment by serveral1** [comment]
I only have SY installed, so I doubt it's an app conflict issue. Either way I'm a little apprehensive on trying to replicate the issue anymore on my end since it's a bit annoying to get all my entries back in order.

**Comment by Macx6** [comment]
Now I have only SY, but I tried other versions and uninstalled them after. Also removing and reinstalling extensions didn't fix the problem. 
For now I just update whole library as many entries have irregular update cycle and it's irritating to check them separately.

@serveral1 is manual category update working for you without problems?

**Comment by serveral1** [comment]
yeah, triggering the global update manually works for me without a hitch, it's the automated global updates that seem to break everything when the app is closed/sleeping.

**Comment by Macx6** [comment]
I haven't noticed this recently, so I think it is resolved in some way.

**Comment by AntsyLich** [comment]
https://github.com/mihonapp/mihon/issues/647
